.github/workflows/validate-migration.yaml

```diff
@@ -57,7 +115,47 @@ jobs:
       - name: "Install c++ compilation dependencies"
         run: |
           sudo apt-get update
-          sudo apt-get install git make cmake gcc g++ clang --yes # p4-fusion dependencies
+          sudo apt-get install --yes \
+            git \
+            make \
+            cmake \
+            gcc \
+            g++ \
+            clang \
+            libc6-dbg \
+            libc++-dev \
+            libc++abi-dev \
+            llvm \
+            ninja-build \
+            gettext
+          sudo apt remove valgrind --yes # remove any existing valgrind installation since we're compiling it from source
```

Comment:
What is the rationale for compiling `valgrind` from source instead of using the pre-built version? The one in Ubuntu 22.04 is about 2 years old, but that should be fine.

Reply:
Sorry, found this answered below.